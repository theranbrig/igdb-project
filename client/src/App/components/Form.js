import React, { useState } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const FormStyles = styled.div`
  width: 90%;
  margin: 0 auto;
  label,
  input,
  button,
  select,
  option {
    font-family: 'Press Start 2p';
  }
  button {
    margin-top: 10px;
  }
`;

const Form = props => {
  const [platform, setPlatform] = useState(props.platform || '');

  const changePlatform = e => {
    setPlatform(e.target.value);
  };

  return (
    <FormStyles>
      <form>
        <label htmlFor="platform" className="nes-text is-primary">
          Choose Your Platform Type
        </label>
        <div className="nes-select">
          <select
            required
            id="platform"
            name="platform"
            onChange={platform => changePlatform(platform)}
            defaultValue={platform}
          >
            <option value="18">Nintendo</option>
            <option value="19">Super Nintendo</option>
            <option value="4">N64</option>
            <option value="21">Game Cube</option>
          </select>
        </div>
        <Link to={{ pathname: '/random', query: { platform } }}>
          <button type="button" className="nes-btn is-success">
            Random Game
          </button>
        </Link>
      </form>
    </FormStyles>
  );
};

Form.propTypes = {
  platform: PropTypes.number,
};

export default Form;
