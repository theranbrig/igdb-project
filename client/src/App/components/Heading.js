import React from 'react';
import { HeadingStyles } from '../styles/MainStyles';

const Heading = () => (
	<HeadingStyles>
		<h1>
			<i className="nes-icon nes-mario is-small" /> NintenDB{' '}
			<i className="nes-icon nes-kirby is-small" />
		</h1>
	</HeadingStyles>
);

export default Heading;
