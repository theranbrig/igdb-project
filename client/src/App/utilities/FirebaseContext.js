import React, { useState } from 'react';
import firebase from 'firebase';
import PropTypes from 'prop-types';

export const FirebaseContext = React.createContext();

const config = {
  apiKey: process.env.REACT_APP_API_KEY,
  authDomain: process.env.REACT_APP_AUTH_DOMAIN,
  databaseURL: process.env.REACT_APP_DATABASE_URL,
  projectId: process.env.REACT_APP_PROJECT_ID,
  storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
  messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
};

firebase.initializeApp(config);

const FirebaseProvider = props => {
  const [error, setError] = useState(null);
  const [authUser, setAuthUser] = useState(null);
  const [loading, setLoading] = useState(false);
  const [liked, setLiked] = useState(false);
  const [savedGames, setSavedGames] = useState([]);
  const [likedGame, setLikedGame] = useState(null);

  const db = firebase.firestore();

  const handleSignUp = async (email, password) => {
    setLoading(true);
    const user = await firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(setLoading(false))
      .catch(function(error) {
        setError(error.message);
        setLoading(false);
      });
    setAuthUser(user);
  };

  const handleSignIn = async (email, password) => {
    setLoading(true);
    const user = await firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(setLoading(false))
      .catch(function(error) {
        setError(error.message);
        setLoading(false);
      });
    setAuthUser(user);
  };

  const handleLogout = async () => {
    setLoading(true);
    await firebase.auth().signOut();
    setAuthUser(null);
    setLoading(false);
  };

  firebase.auth().onAuthStateChanged(function(user) {
    if (user) {
      setAuthUser(user);
    } else {
      console.log('No User Found');
    }
  });

  const checkGameSave = async (userId, gameId) => {
    setLiked(false);
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .where('gameId', '==', gameId)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          if (doc.data()) {
            setLiked(true);
            setLikedGame(doc.data().likeId);
          }
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };

  const handleGameSave = async (userId, gameId, name, platform) => {
    await db
      .collection('votes')
      .add({
        userId,
        gameId,
        name,
        platform,
      })
      .then(async function(docRef) {
        await checkGameSave(userId, docRef.id);
      })
      .catch(function(error) {
        console.error('Error adding document: ', error);
      });
  };

  const handleGameUnsave = async (userId, gameId) => {
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .where('gameId', '==', gameId)
      .get()
      .then(function(querySnapshot) {
        querySnapshot.forEach(function(doc) {
          doc.ref.delete();
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
  };

  const checkSavedUserGames = async userId => {
    const tempSavedGames = [];
    await db
      .collection('votes')
      .where('userId', '==', userId)
      .get()
      .then(querySnapshot => {
        querySnapshot.forEach(function(doc) {
          if (doc.data()) {
            tempSavedGames.push(doc.data());
          }
        });
      })
      .catch(function(error) {
        console.log('Error getting documents: ', error);
      });
    setSavedGames(tempSavedGames);
  };

  return (
    <FirebaseContext.Provider
      value={{
        error,
        authUser,
        handleSignUp,
        handleSignIn,
        handleLogout,
        loading,
        handleGameSave,
        checkGameSave,
        checkSavedUserGames,
        liked,
        savedGames,
        setLoading,
        likedGame,
        handleGameUnsave,
        setLiked,
      }}
    >
      {props.children}
    </FirebaseContext.Provider>
  );
};

FirebaseProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default FirebaseProvider;
