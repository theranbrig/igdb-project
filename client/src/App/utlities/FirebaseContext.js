import React, { useState } from 'react';
import firebase from 'firebase';

export const FirebaseContext = React.createContext();

const config = {
	apiKey: process.env.REACT_APP_API_KEY,
	authDomain: process.env.REACT_APP_AUTH_DOMAIN,
	databaseURL: process.env.REACT_APP_DATABASE_URL,
	projectId: process.env.REACT_APP_PROJECT_ID,
	storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
	messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID
};

firebase.initializeApp(config);

const FirebaseProvider = props => {
	const [error, setError] = useState(null);
	// Auth User = authUser.uid /
	const [authUser, setAuthUser] = useState(null);
	const [loading, setLoading] = useState(false);
	const [liked, setLiked] = useState(false);
	const [savedGames, setSavedGames] = useState([]);

	const db = firebase.firestore();

	const handleSignUp = async (email, password) => {
		setLoading(true);
		const user = await firebase
			.auth()
			.createUserWithEmailAndPassword(email, password)
			.then(() => db.collection('users').add({ email }))
			.then(docRef => {
				db.collection('users')
					.doc(docRef.id)
					.update({ userId: docRef.id });
				setLoading(false);
			})
			.catch(function(error) {
				setError(error.message);
				setLoading(false);
			});
		setAuthUser(user);
	};

	const handleSignIn = async (email, password) => {
		setLoading(true);
		const user = await firebase
			.auth()
			.signInWithEmailAndPassword(email, password)
			.then(setLoading(false))
			.catch(function(error) {
				setError(error.message);
				setLoading(false);
			});
		setAuthUser(user);
	};

	const handleLogout = async () => {
		setLoading(true);
		await firebase.auth().signOut();
		setAuthUser(null);
		setLoading(false);
	};

	firebase.auth().onAuthStateChanged(function(user) {
		if (user) {
			setAuthUser(user);
		} else {
			console.log('No User Found');
		}
	});

	const handleGameSave = async (userId, gameId, name, platform) => {
		const like = await db
			.collection('votes')
			.add({
				userId,
				gameId,
				name,
				platform
			})
			.then(function(docRef) {
				console.log('Document written with ID: ', docRef.id);
			})
			.catch(function(error) {
				console.error('Error adding document: ', error);
			});
	};

	const checkGameSave = async (userId, gameId) => {
		setLiked(false);
		const saveCheck = await db
			.collection('votes')
			.where('userId', '==', userId)
			.where('gameId', '==', gameId)
			.get()
			.then(function(querySnapshot) {
				querySnapshot.forEach(function(doc) {
					if (doc.data()) {
						setLiked(true);
					}
				});
			})
			.catch(function(error) {
				console.log('Error getting documents: ', error);
			});
	};

	const checkSavedUserGames = async userId => {
		const tempSavedGames = [];
		const response = await db
			.collection('votes')
			.where('userId', '==', userId)
			.get()
			.then(querySnapshot => {
				querySnapshot.forEach(function(doc) {
					if (doc.data()) {
						tempSavedGames.push(doc.data());
					}
				});
			})
			.catch(function(error) {
				console.log('Error getting documents: ', error);
			});
		setSavedGames(tempSavedGames);
	};

	return (
		<FirebaseContext.Provider
			value={{
				error,
				authUser,
				handleSignUp,
				handleSignIn,
				handleLogout,
				loading,
				handleGameSave,
				checkGameSave,
				checkSavedUserGames,
				liked,
				savedGames
			}}>
			{props.children}
		</FirebaseContext.Provider>
	);
};

export default FirebaseProvider;
