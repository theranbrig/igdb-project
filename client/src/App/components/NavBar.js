import React, { useContext } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { FirebaseContext } from '../utlities/FirebaseContext';

const NavStyles = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  height: 100px;
  .nav-logo,
  .nav-links {
    display: inline-flex;
    align-self: center;
    p {
      font-size: 2rem;
      margin-left: 10px;
    }
    a {
      display: inherit;
      color: black;
      font-family: 'Press Start 2p';
      &:hover {
        text-decoration: none;
        text-shadow: 1px 1px 10px gray;
      }
    }
  }
  .nav-logo {
    margin-left: 10px;
  }
  .nav-links button {
    margin: 10px;
    font-family: 'Press Start 2p';
  }
`;

const NavBar = () => {
  const { authUser } = useContext(FirebaseContext);
  return (
    <NavStyles>
      <div className="nav-logo">
        <Link to="/">
          <i className="nes-logo" />
          <p>NintenDB</p>
        </Link>
      </div>
      <div className="nav-links">
        <Link to={{ pathname: '/random', query: { platform: 18 } }}>
          <button className="nes-btn is-success">Random Game</button>
        </Link>
        {authUser ? (
          <Link to="/mypage">
            <button className="nes-btn is-primary">My Saved Games</button>
          </Link>
        ) : (
          <>
            <Link to="/signup">
              <button className="nes-btn is-primary">Sign Up</button>
            </Link>
            <Link to="/login">
              <button className="nes-btn is-primary">Login</button>
            </Link>
          </>
        )}
      </div>
    </NavStyles>
  );
};

export default NavBar;
