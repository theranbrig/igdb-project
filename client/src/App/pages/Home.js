import styled from 'styled-components';
import React from 'react';
import Form from '../components/Form';
import Heading from '../components/Heading';
import NavBar from '../components/NavBar';

const MainStyles = styled.div`
	.App {
		margin: 20px;
		min-height: 80vh;
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		margin: 10vh auto;
		width: 90%;
		.title {
			font-family: 'Press Start 2p';
		}
	}
	h1,
	i {
		padding: 15px 0 0;
		font-family: 'Press Start 2p';
		font-size: 5rem;
		text-align: center;
	}
	p {
		font-family: 'Press Start 2p';
	}
	.loading-screen {
		display: grid;
		grid-template-rows: 1fr;
		grid-template-columns: 1fr;
		min-height: 80vh;
		align-items: center;
		justify-items: center;
		margin: 10vh auto;
	}
	.icon-list {
		margin: 30px;
		i {
			margin: 20px !important;
		}import Heading from './Heading';
import NavBar from '../components/NavBar';

	}
`;

const Home = () => (
  <MainStyles>
    <NavBar />
    <div className="App nes-container is-dark is-rounded">
      <Heading />
      <Form />
      <section className="icon-list">
        <i className="nes-ash" />
        <i className="nes-pokeball" />
        <i className="nes-bulbasaur" />
        <i className="nes-charmander" />
        <i className="nes-squirtle" />
      </section>
    </div>
  </MainStyles>
);

export default Home;
