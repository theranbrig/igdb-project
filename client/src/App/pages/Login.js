import React, { useState, useContext } from 'react';
import styled from 'styled-components';
import { withRouter, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import NavBar from '../components/NavBar';
import { FirebaseContext } from '../utilities/FirebaseContext';
import Spinner from '../components/Spinner';

const SignUpFormStyles = styled.div`
  text-align: center;
  label,
  input,
  button,
  select,
  option,
  h1,
  p {
    font-family: 'Press Start 2p';
  }
  label {
    width: 100%;
  }

  input {
    margin-top: 10px;
  }
  form {
    margin: 0 auto;
    display: grid;
    grid-template-rows: 1fr 1fr 1fr;
    width: 50%;
    grid-gap: 20px;
  }
`;

const SignUpForm = ({ history }) => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const { handleSignIn, error, loading } = useContext(FirebaseContext);

  return (
    <SignUpFormStyles>
      <NavBar />
      <h1>Login</h1>
      {loading ? (
        <Spinner loading={loading} />
      ) : (
        <form
          onSubmit={async e => {
            e.preventDefault();
            await handleSignIn(email, password);
            if (!error) {
              history.push('/');
            }
          }}
        >
          <label>
            Email
            <input
              onChange={e => setEmail(e.target.value)}
              type="text"
              value={email}
              placeholder="Enter Email"
              className="nes-input"
              autoComplete="email"
            />
          </label>
          <label>
            Password
            <input
              onChange={e => setPassword(e.target.value)}
              autoComplete="password"
              type="password"
              value={password}
              placeholder="Enter Password"
              className="nes-input"
            />
          </label>

          <button className="nes-btn is-primary" type="submit" disabled={email === '' && password}>
            Sign Up
          </button>
          <Link to="/signup">
            <p className="nes-text is-primary">Not yet a member? Sign Up Today.</p>
          </Link>
          {error && <p>{error}</p>}
        </form>
      )}
    </SignUpFormStyles>
  );
};

SignUpForm.propTypes = {
  history: PropTypes.object.isRequired,
};

export default withRouter(SignUpForm);
