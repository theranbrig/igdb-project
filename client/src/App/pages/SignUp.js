import React, { useState, useContext } from 'react';
import styled from 'styled-components';
import { withRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';
import { FirebaseContext } from '../utilities/FirebaseContext';

const SignUpFormStyles = styled.div`
  text-align: center;
  label,
  input,
  button,
  select,
  option,
  h1,
  p {
    font-family: 'Press Start 2p';
  }
  label {
    width: 100%;
  }

  input {
    margin-top: 10px;
  }
  form {
    margin: 0 auto;
    display: grid;
    grid-template-rows: 1fr 1fr 1fr;
    width: 50%;
    grid-gap: 20px;
  }
`;

const SignUpForm = () => {
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  const { handleSignUp, error } = useContext(FirebaseContext);

  return (
    <SignUpFormStyles>
      <NavBar />
      <h1>Sign Up Today</h1>
      <form
        onSubmit={async e => {
          e.preventDefault();
          await handleSignUp(email, password);
        }}
      >
        <div className="nes-field">
          <label>
            Email
            <input
              onChange={e => setEmail(e.target.value)}
              type="text"
              id="email"
              value={email}
              placeholder="Enter Email"
              className="nes-input"
              autoComplete="email"
            />
          </label>
        </div>
        <div className="nes-field">
          <label>
            Password
            <input
              onChange={e => setPassword(e.target.value)}
              type="password"
              id="password"
              value={password}
              placeholder="Enter Password"
              className="nes-input"
              autoComplete="password"
            />
          </label>
        </div>
        <button className="nes-btn is-primary" type="submit">
          Sign Up
        </button>
        {error && <p>{error}</p>}
      </form>
    </SignUpFormStyles>
  );
};

export default withRouter(SignUpForm);
