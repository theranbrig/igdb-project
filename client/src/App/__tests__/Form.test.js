import React from 'react';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Heading from '../components/Heading';
import Home from '../pages/Home';
import { FirebaseContext } from '../utilities/FirebaseContext';
import NavBar from '../components/NavBar';
import Form from '../components/Form';
import SignUp from '../pages/SignUp';
import Login from '../pages/Login';

const authUser = { email: 'fake@fake.com' };
const nonAuthUser = null;
const platform = 18;

it('renders the platfrom form correctly', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ platform }}>
				<Form />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});

it('renders the signup form correctly', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ platform }}>
				<SignUp />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});

it('renders the login form correctly', () => {
	const element = TestRenderer.create(
		<Router>
			<FirebaseContext.Provider value={{ platform }}>
				<Login />
			</FirebaseContext.Provider>
		</Router>
	).toJSON();
	expect(element).toMatchSnapshot();
});
