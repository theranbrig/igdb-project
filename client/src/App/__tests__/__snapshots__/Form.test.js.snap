// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the login form correctly 1`] = `
<div
  className="sc-ifAKCX jEbIyP"
>
  <div
    className="sc-htpNat jdiiWr"
  >
    <div
      className="nav-logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="nes-logo"
        />
        <p>
          NintenDB
        </p>
      </a>
    </div>
    <div
      className="nav-links"
    >
      <a
        href="/random"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-success"
          type="button"
        >
          Random Game
        </button>
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-primary"
          type="button"
        >
          Sign Up
        </button>
      </a>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-primary"
          type="button"
        >
          Login
        </button>
      </a>
    </div>
  </div>
  <div
    className="sc-bxivhb bbnBSQ"
  >
    <h1>
      <i
        className="nes-icon nes-mario is-small"
      />
       NintenDB 
      <i
        className="nes-icon nes-kirby is-small"
      />
    </h1>
  </div>
  <h1>
    Login
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      Email
      <input
        autoComplete="email"
        className="nes-input"
        onChange={[Function]}
        placeholder="Enter Email"
        type="text"
        value=""
      />
    </label>
    <label>
      Password
      <input
        autoComplete="password"
        className="nes-input"
        onChange={[Function]}
        placeholder="Enter Password"
        type="password"
        value=""
      />
    </label>
    <button
      className="nes-btn is-primary"
      disabled=""
      type="submit"
    >
      Sign Up
    </button>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <p
        className="nes-text is-primary"
      >
        Not yet a member? Sign Up Today.
      </p>
    </a>
  </form>
</div>
`;

exports[`renders the platfrom form correctly 1`] = `
<div
  className="sc-EHOje bCUIMy"
>
  <form>
    <label
      className="nes-text is-primary"
      htmlFor="platform"
    >
      Choose Your Platform Type
    </label>
    <div
      className="nes-select"
    >
      <select
        defaultValue=""
        id="platform"
        name="platform"
        onChange={[Function]}
        required={true}
      >
        <option
          value="18"
        >
          Nintendo
        </option>
        <option
          value="19"
        >
          Super Nintendo
        </option>
        <option
          value="4"
        >
          N64
        </option>
        <option
          value="21"
        >
          Game Cube
        </option>
      </select>
    </div>
    <a
      href="/random"
      onClick={[Function]}
    >
      <button
        className="nes-btn is-success"
        type="button"
      >
        Random Game
      </button>
    </a>
  </form>
</div>
`;

exports[`renders the signup form correctly 1`] = `
<div
  className="sc-ifAKCX jEbIyP"
>
  <div
    className="sc-htpNat jdiiWr"
  >
    <div
      className="nav-logo"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <i
          className="nes-logo"
        />
        <p>
          NintenDB
        </p>
      </a>
    </div>
    <div
      className="nav-links"
    >
      <a
        href="/random"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-success"
          type="button"
        >
          Random Game
        </button>
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-primary"
          type="button"
        >
          Sign Up
        </button>
      </a>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="nes-btn is-primary"
          type="button"
        >
          Login
        </button>
      </a>
    </div>
  </div>
  <div
    className="sc-bxivhb bbnBSQ"
  >
    <h1>
      <i
        className="nes-icon nes-mario is-small"
      />
       NintenDB 
      <i
        className="nes-icon nes-kirby is-small"
      />
    </h1>
  </div>
  <h1>
    Sign Up Today
  </h1>
  <form
    onSubmit={[Function]}
  >
    <div
      className="nes-field"
    >
      <label>
        Email
        <input
          autoComplete="email"
          className="nes-input"
          id="email"
          onChange={[Function]}
          placeholder="Enter Email"
          type="text"
          value=""
        />
      </label>
    </div>
    <div
      className="nes-field"
    >
      <label>
        Password
        <input
          autoComplete="password"
          className="nes-input"
          id="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
      </label>
    </div>
    <button
      className="nes-btn is-primary"
      type="submit"
    >
      Sign Up
    </button>
    <a
      href="/login"
      onClick={[Function]}
    >
      <p
        className="nes-text is-primary"
      >
        Already a Member? Login Today.
      </p>
    </a>
  </form>
</div>
`;
